pkg/controller/customdomain/customdomain_controller.go

```diff
@@ -374,6 +374,55 @@ func modifyClusterDomain(r *ReconcileCustomDomain, reqLogger logr.Logger, instan
 		return err
 	}
 
+	// modify apiservers.config.openshift.io/cluster to support login on new domain
+	apiserverConfig := &configv1.APIServer{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{
+		Name: "cluster",
+	}, apiserverConfig)
+	if err != nil {
+		reqLogger.Error(err, "Error getting apiservers.config.openshift.io/cluster")
+		// Error reading the object - requeue the request.
+		return err
+	}
+	if !finalize {
+		if _, ok := instance.ObjectMeta.Annotations["original-apiserver-config"]; !ok {
+			encSpec, _ := json.Marshal(apiserverConfig.Spec)
+			instance.ObjectMeta.Annotations["original-apiserver-config"] = string(encSpec)
+			err = r.client.Update(context.TODO(), instance)
+			if err != nil {
+				reqLogger.Error(err, "Error updating instance with 'original-apiserver-config' annotation")
+			}
+		}
+		found := false
+		for _, item := range apiserverConfig.Spec.ServingCerts.NamedCertificates {
+			if contains(item.Names, "api."+domain) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			namedCert := configv1.APIServerNamedServingCert{
+				Names: []string{string("api." + domain)},
+				ServingCertificate: configv1.SecretNameReference{
+					Name: cert,
+				},
+			}
+			apiserverConfig.Spec.ServingCerts.NamedCertificates = append(apiserverConfig.Spec.ServingCerts.NamedCertificates, namedCert)
+		}
+	} else {
+		// deserialize original apiserver-config
+		if _, ok := instance.ObjectMeta.Annotations["original-apiserver-config"]; ok {
```

Comment:
If a customer makes a typo adding the second domain and they edit it will the typo'd domain be added to original-apiserver-config or can it not be editied?

Reply:
If I understand your question correctly, you're asking about more than one CustomDomain CR? This is assuming that there is only one CustomDomain CR at a time. The original-* annotations are intended to hold the serialized specs of the original cluster specs before the CustomDomain CR was applied. I'll create a ticket/issue to ensure that there is only one CR at atime.